sdk/schemaregistry/azure-data-schemaregistry/src/main/java/com/azure/data/schemaregistry/implementation/SchemaRegistryHelper.java

```diff
@@ -38,33 +52,80 @@ public static void setAccessor(SchemaRegistryModelsAccessor modelsAccessor) {
         final SchemasGetByIdHeaders headers = response.getDeserializedHeaders();
+        final SchemaFormat schemaFormat = getSchemaFormat(response.getHeaders());
 
-        return accessor.getSchemaProperties(headers.getSchemaId(), SchemaFormat.AVRO, headers.getSchemaGroupName(),
+        return accessor.getSchemaProperties(headers.getSchemaId(), schemaFormat, headers.getSchemaGroupName(),
             headers.getSchemaName(), headers.getSchemaVersion());
     }
 
-    public static SchemaProperties getSchemaProperties(SchemasQueryIdByContentResponse response) {
+    public static SchemaProperties getSchemaProperties(SchemasQueryIdByContentResponse response,
+        SchemaFormat fallbackFormat) {
+
         final SchemasQueryIdByContentHeaders headers = response.getDeserializedHeaders();
+        final SchemaFormat responseFormat = getSchemaFormat(response.getHeaders());
+        final SchemaFormat schemaFormat = responseFormat != null ? responseFormat : fallbackFormat;
 
-        return accessor.getSchemaProperties(headers.getSchemaId(), SchemaFormat.AVRO, headers.getSchemaGroupName(),
+        return accessor.getSchemaProperties(headers.getSchemaId(), schemaFormat, headers.getSchemaGroupName(),
             headers.getSchemaName(), headers.getSchemaVersion());
     }
 
     public static SchemaProperties getSchemaProperties(SchemasGetSchemaVersionResponse response) {
         final SchemasGetSchemaVersionHeaders headers = response.getDeserializedHeaders();
+        final SchemaFormat schemaFormat = getSchemaFormat(response.getHeaders());
 
-        return accessor.getSchemaProperties(headers.getSchemaId(), SchemaFormat.AVRO, headers.getSchemaGroupName(),
+        return accessor.getSchemaProperties(headers.getSchemaId(), schemaFormat, headers.getSchemaGroupName(),
             headers.getSchemaName(), headers.getSchemaVersion());
     }
 
+    public static SchemaFormatImpl getContentType(SchemaFormat schemaFormat) {
+        Objects.requireNonNull(schemaFormat, "'schemaFormat' cannot be null.'");
+
+        if (schemaFormat == SchemaFormat.AVRO) {
+            return SchemaFormatImpl.APPLICATION_JSON_SERIALIZATION_AVRO;
+        } else if (schemaFormat == SchemaFormat.JSON) {
+            return SchemaFormatImpl.APPLICATION_JSON_SERIALIZATION_JSON;
+        } else {
+            return SchemaFormatImpl.TEXT_PLAIN_CHARSET_UTF8;
+        }
+    }
+
+    /**
+     * Extracts "Content-Type" from HttpHeaders and translates it into {@link SchemaFormat}.
+     *
+     * @param headers Headers to read.
+     * @return The corresponding {@link SchemaFormat} or {@code null} if the header does not exist.
+     */
+    public static SchemaFormat getSchemaFormat(HttpHeaders headers) {
+        final String contentType = headers.getValue("Content-Type");
+
+        if (contentType == null) {
+            return null;
+        }
+
+        final String replaced = contentType.replaceAll("\\s", "").toLowerCase(Locale.ROOT);
```

Comment:
This can be an expensive operation if its invoked per API call.
Internally it will call Pattern.compile("\\s").matcher(..)
You can create the Pattern as a static variable once and reuse it per API call.